src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -339,4 +358,59 @@ private function announce($subject, Transition $initialTransition, Marking $mark
             $this->dispatcher->dispatch(sprintf('workflow.%s.announce.%s', $this->name, $transition->getName()), $event);
         }
     }
+
+    /**
+     * @param string $transitionName
+     *
+     * @return Transition[]
+     */
+    private function getTransitionsByName(string $transitionName)
+    {
+        $transitions = array_filter(
+            $this->definition->getTransitions(),
+            function (Transition $transition) use ($transitionName) {
+                return $transition->getName() === $transitionName;
+            }
+        );
+
+        return $transitions;
+    }
+
+    /**
+     * Returns all enabled transitions or a transition blocker list of one of them.
+     *
+     * @param object $subject        A subject
+     * @param string $transitionName
+     *
+     * @return Transition[]|TransitionBlockerList All enabled transitions or a blocker list
+     *                                            if no enabled transitions can be found
+     */
+    private function getEnabledTransitionsByNameOrTransitionBlockerList($subject, string $transitionName)
+    {
+        $eligibleTransitions = $this->getTransitionsByName($transitionName);
+
+        if (!$eligibleTransitions) {
+            return new TransitionBlockerList(array(TransitionBlocker::createNotDefined($transitionName, $this->name)));
+        }
+
+        $marking = $this->getMarking($subject);
+        $transitions = array();
+
+        // this is needed to silence static analysis in phpstorm
+        $transitionBlockerList = new TransitionBlockerList();
```

Comment:
Latest phpstorm can't understand, that at this point there must be at least 1 `$eligibleTransitions`. And I don't like static inspector warnings in source files. If you'd like to remove this line, then I'm fine with that.